src/Symfony/Bundle/FrameworkBundle/Tests/Functional/ContainerDebugCommandTest.php

```diff
@@ -136,6 +136,82 @@ public function testDescribeEnvVar()
         $this->assertContains(file_get_contents(__DIR__.'/Fixtures/describe_env_vars.txt'), $tester->getDisplay(true));
     }
 
+    public function testGetDeprecation()
+    {
+        static::bootKernel(['test_case' => 'ContainerDebug', 'root_config' => 'config.yml', 'debug' => true]);
+        $path = sprintf('%s/%sDeprecations.log', static::$kernel->getContainer()->getParameter('kernel.cache_dir'), static::$kernel->getContainer()->getParameter('kernel.container_class'));
+        touch($path);
+        file_put_contents($path, 'a:2:{i:0;a:6:{s:4:"type";i:16384;s:7:"message";s:171:"The "Symfony\Bundle\FrameworkBundle\Controller\Controller" class is deprecated since Symfony 4.2, use Symfony\Bundle\FrameworkBundle\Controller\AbstractController instead.";s:4:"file";s:87:"/home/hamza/projet/contrib/sf/vendor/symfony/framework-bundle/Controller/Controller.php";s:4:"line";i:17;s:5:"trace";a:1:{i:0;a:3:{s:4:"file";s:66:"/home/hamza/projet/contrib/sf/src/Controller/DefaultController.php";s:4:"line";i:9;s:8:"function";s:17:"spl_autoload_call";}}s:5:"count";i:1;}i:1;a:6:{s:4:"type";i:16384;s:7:"message";s:188:"The "App\Controller\DefaultController" class extends "Symfony\Bundle\FrameworkBundle\Controller\Controller" that is deprecated since Symfony 4.2, use {@see AbstractController} instead.
```

Comment:
for sanity i would use `serialize(array(...))` here :) 

can we generate a real deprecation? that be even better

Reply:
this is just to mock the deprecation, I guess we are testing that a deprecation is generated :p.

Reply:
but we test a feature that relies upon a deprecation being triggered :) the current approach works for me (though `serialize(array())` would ease - future - maintenance)

duplicating the log format here introduces double bookkeeping (we may need to update the mock if the implementation is update - which we usually forget unless it breaks :sweat_smile: )